### PR TITLE
First try... GroupNumber1

### PR DESCRIPTION
**Pull request by Harzenetter**
First attempt to rename like clean code style... perhaps

### CODE REVIEW

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 1]
```diff
@@ -4,28 +4,28 @@
...
-		return builder.toString();
+		return diamondBuilder.toString();
 	}
 
-	private char[] make(int i, int j) {
-		int amount = ((i - j) / 2);
-		char[] chars = new char[amount + j + 1];
-		if (amount > 0) {
-			Arrays.fill(chars, 0, amount, ' ');
+	private char[] createDiamondRow(int maximumDimension, int sizeToLower) {
```
Comment (corebonts) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:20]: Group4: sizeToLower is not that obvious in this context

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 2]
```diff
@@ -4,28 +4,28 @@
...
+		int amountOfStars = ((maximumDimension - sizeToLower) / 2);
```
Comment (corebonts) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:21]: This is the padding and not the amount of the stars.

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 3]
```diff
@@ -4,28 +4,28 @@
 
 public class Diamond {
 
-	public String print(int n) {
-		if (n <= 0 || n % 2 == 0) {
+	public String createDiamond(int maximumDimension) {
```
Comment (corebonts) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:7]: createDiamond: It's debatable whether this API change worth it. this name feels for us like a factory method.
maximumDimension: Is it only a restriction? Can the method create a smaller diamond as well?

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 4]
```diff
@@ -4,28 +4,28 @@
...
+		if (maximumDimension <= 0 || maximumDimension % 2 == 0) {
```
Comment (zopad) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:8]: Consider moving this check into an isValidDiamond() method just to be self-documenting.

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 5]
```diff
@@ -4,28 +4,28 @@
...
 			return null;
 		}
-		StringBuilder builder = new StringBuilder(new String(make(n, n)));
-		for (int i = n - 2; i > 0; i -= 2) {
-			char[] chars = make(n, i);
-			builder.insert(0, chars);
-			builder.append(chars);
+		StringBuilder diamondBuilder = new StringBuilder(new String(createDiamondRow(maximumDimension, maximumDimension)));
+		for (int sizeToLower = maximumDimension - 2; sizeToLower > 0; sizeToLower -= 2) {
```
Comment (zopad) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:12]: sizeToLower would be better as numberOfStars or similar.

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java [hunk 6]
```diff
@@ -4,28 +4,28 @@
...
+	public String createDiamond(int maximumDimension) {
+		if (maximumDimension <= 0 || maximumDimension % 2 == 0) {
 			return null;
 		}
-		StringBuilder builder = new StringBuilder(new String(make(n, n)));
-		for (int i = n - 2; i > 0; i -= 2) {
-			char[] chars = make(n, i);
-			builder.insert(0, chars);
-			builder.append(chars);
+		StringBuilder diamondBuilder = new StringBuilder(new String(createDiamondRow(maximumDimension, maximumDimension)));
```
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:11]: `Builder` is redundant

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java (hunk 4 above)
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:8]: Maybe factor out?

File: src/main/java/de/cas/mse/exercise/diamond/Diamond.java (hunk 5 above)
Comment (TehMillhouse) [inline src/main/java/de/cas/mse/exercise/diamond/Diamond.java:12]: `sizeToLower` doesn't make sense to me tbh

**Pull request merged**